### DISCUSSION TITLE
how to change  set(CTEST_DROP_METHOD "https") to http on production docker instance

### DISCUSSION THREAD

**Discussion by samikh2020** [Kitware/CDash#2028] [General]
Apart of our infrastructure we have some instances that are unable to submit throught https - is there a way of changing the drop method to http?

As part of this ive extended the docker compose files to be open on port 443 and 8080 and updated the apache config aswell.

**Comment by samikh2020** [comment]
is it possible to run the production docker instance without https?

**Comment by williamjallen** [comment]
You can set `APP_ENV` to `development`.

**Comment by samikh2020** [comment]
Thanks @williamjallen,

ive tried to set app_env to developement however it makes GUI not work - im able to see the top banner and login but nothing passed that - ive tried on both chrome and firefox.

my env file looks like this:
```
# See app.php in the config/ directory for a description of these variables.
APP_NAME=CDash
APP_ENV=development
#APP_DEBUG=true
APP_KEY=keMCcoZh8NaZ7jfSOGbAgicX2Ch2E9iy
APP_URL=http://cdash-staging.team.aws.company.com
MIX_APP_URL="${APP_URL}"

# The following two variables are only used for docker compose production installations.
SSL_CERTIFICATE_FILE=/home/ubuntu/CDash/ssl/fullchain.pem
SSL_CERTIFICATE_KEY_FILE=/home/ubuntu/CDash/ssl/key.pem

# database.php
DB_DATABASE=cdash
DB_PASSWORD=cdash

QUEUE_CONNECTION=database
```

and im using this command to launch the instance

```
docker compose --env-file .env \
	   -f docker/docker-compose.yml \
	   -f docker/docker-compose.production.yml \
	   -f docker/docker-compose.mysql.yml \
	    up -d
```

**Comment by samikh2020** [comment]
when i use https in the app_url - i do seem to resolve the GUI issues but then the   set(CTEST_DROP_METHOD "https") is auto set to https also

**Comment by samikh2020** [comment]
my ideal solution would be to have https on access the GUI and be able to receive HTTP submissions

**Comment by williamjallen** [comment]
Did you run `npm run prod` after switching `APP_ENV` to `development`?

**Comment by samikh2020** [comment]
i destroyed the containers and volumes and launched fresh

**Comment by zackgalbreath** [comment]
> my ideal solution would be to have https on access the GUI and be able to receive HTTP submissions

I don't think there's currently a "built-in" way to achieve this with CDash's docker images, but in the past we've achieved this with careful use of apache RewriteRules. Let me see if I can dig up an example for you.

**Reply by zackgalbreath** [reply]
```xml
# The sole purpose of this VirtualHost is to submissions over http (not https).
# You'll still need a separate VirtualHost serving CDash traffic on port 443. 

<VirtualHost *:80>
ServerName cdash.mydomain.com
DocumentRoot /var/www/cdash
<Directory /var/www/cdash>
     AllowOverride All
     Options -Indexes
</Directory>

RewriteEngine On
RewriteCond %{HTTPS} off
RewriteCond %{REQUEST_URI} !^\/submit\.php
RewriteCond %{REQUEST_URI} !^\/index\.php
RewriteRule (.*) https://cdash.mydomain.com%{REQUEST_URI} [R=301,L]

</VirtualHost>
```
